omaha/v3/protocol_test.go

```diff
@@ -0,0 +1,572 @@
+package v3
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"strings"
+	"testing"
+
+	"github.com/brave/go-update/extension"
+)
+
+func TestNewProtocol(t *testing.T) {
+	tests := []struct {
+		name        string
+		version     string
+		wantVersion string
+	}{
+		{
+			name:        "Version 3.0",
+			version:     "3.0",
+			wantVersion: "3.0",
+		},
+		{
+			name:        "Version 3.1",
+			version:     "3.1",
+			wantVersion: "3.1",
+		},
+		{
+			name:        "Any other version",
+			version:     "4.0",
+			wantVersion: "4.0", // Will now just store the version
```

Comment:
What happens here w/ the test? It should be erroring because 4 is not supported, right?